blog-posts/from-promises-to-observables/from-promises-to-observables.md

```diff
@@ -0,0 +1,390 @@
+
+```ts
+  async load() {
+    // show spinner while fetching posts
+    this.showSpinner = true;
+
+    // retrieve the posts by calling the endpoint that uses promise.all for fetching
+    // all of the rss feeds and waiting (synchronously) for them to complete
+    this.posts = [];
+    const response = await axios.get(environment.traditionalEndpoint);
+    response.data.forEach((entry) => {
+      const inputDate = new Date(entry.pubDate);
+      entry.pubDate = inputDate.toLocaleDateString('en-us') + ' at ' + inputDate.toLocaleTimeString('en-us');
+      
+      this.posts.push({
+        ...entry, 
+        sortDate: inputDate.getTime()
+      });
+    });
+    this.posts = response.data;
+
+    // retrieve the manual entries
+    const manualEntries: any
+      = await axios.get(environment.manualEntries);
+    manualEntries.data.forEach((entry: any) => {
+      const inputDate = new Date(entry.pubDate);
+      entry.pubDate = inputDate.toLocaleDateString('en-us') + ' at ' + inputDate.toLocaleTimeString('en-us');
+      if (entry.contentSnippet.length > 200) {
+        entry.contentSnippet = entry.contentSnippet.substring(0, 200);
+      }
+
+      this.posts.push({
+        ...entry, 
+        sortDate: inputDate.getTime()
+      });
+    });
+
+    // sort by date here
+    this.posts.sort((a: any, b: any) => {
+      return b.sortDate - a.sortDate;
+    });
+
+    // stop showing spinner when fetch completes
+    this.showSpinner = false;
+  }
+```
+
+Here I've created a `load` method that uses [axios](https://www.npmjs.com/package/axios) to make a GET call to retrieve the data.  I also call an endpoint for an additional set of manual entries.  When both complete I stop showing the spinner.
+
+# Enter RxJS
+
+So after my experiences from Angular Denver, I started thinking of ways that RxJS could improve this whole setup.  I really didn't like the spinner and several second wait time, so thought this was a great opportunity to improve the site.
+
+I decided that it would help if I could break down the HTTP calls to be handled individually as streams.  Then the user would see results immediately on their page, and it would update as the calls completed.  Since its only a matter of seconds, this didn't make the page jump around too much and made the site feel very responsive.
+
+> I should note that typically with frontend engineering you don't want a page to jump around.  In this situation it was only a matter of seconds and the way that I built the flow made it hardly noticeable.  It was more important for me to show the output immediately.  In a larger application this approach may need to be reworked for a better experience.
+
+I refactored the HTTP calls to be done in one endpoint.  If you look in the example project in the `functions/index.js` file you will see this as the following:
+
+```js
+app.get('/reactive/:source', (req, res) => {
```

Comment:
I would consider removing this code example. You can leave your note above and maybe add an info to check out file... if one is interested in the code, but this is not really rxjs related and therefore the reader might lose focus